### PR TITLE
Improvements in Makefile, Dockerfile, added blackfire

### PR DESCRIPTION
**Pull request by rpg600**
This pr does: 
- merge the two dockerfiles into one, the tools container inherit the app container.
- all commands in the makefile are executed in the tools one.
- use the doctrine migrations commands in the makefile
- add blackfire with documentation
- the paybox parameters are moved in the dev config file.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -20,41 +21,48 @@ deps:           ## [composer yarn assets-dev perm] Install the project PHP and J
...
+	$(CONSOLE) doctrine:migrations:migrate -n
 
 db-fixtures:    ## Reloads the data fixtures for the dev environment
 	$(CONSOLE) doctrine:fixtures:load -n
 
 db-update:      ## Update the database structure according to the last changes
-	$(CONSOLE) doctrine:schema:update --force
+	$(CONSOLE) doctrine:migration:migrate -n
+
+db-diff:
```
Comment (tgalopin) [inline Makefile]: The comment of other commands (`## Update the database structure according to the last changes`) is used to generate the help. Can you add one here :) ?

File: Dockerfile [hunk 2]
```diff
@@ -55,8 +57,25 @@ RUN apt-get update -q && \
 
     curl -sS https://getcomposer.org/installer | php -- --install-dir=/usr/local/bin --filename=composer && \
 
+    curl -L -o /tmp/nodejs.tar.gz https://nodejs.org/dist/v${NODE_VERSION}/node-v${NODE_VERSION}-linux-x64.tar.gz && \
```
Comment (tgalopin) [inline Dockerfile]: Why adding nodeJS to the app? Isn't the tools container made for this?

File: Dockerfile [hunk 3]
```diff
@@ -55,8 +57,25 @@ RUN apt-get update -q && \
...
+    tar xfvz /tmp/nodejs.tar.gz -C /usr/local --strip-components=1 && \
+    rm -f /tmp/nodejs.tar.gz && \
+    npm install yarn -g && \
+
     mkdir /run/php
 
+# Blackfire
```
Comment (tgalopin) [inline Dockerfile]: Nice

**Comment by hhamon** [comment]
👍

File: Makefile [hunk 4]
```diff
@@ -20,41 +21,48 @@ deps:           ## [composer yarn assets-dev perm] Install the project PHP and J
...
+	$(CONSOLE) doctrine:migration:diff
+
+db-reset: db-create db-fixtures
```
Comment (tgalopin) [inline Makefile]: Same

File: Makefile [hunk 5]
```diff
@@ -20,41 +21,48 @@ deps:           ## [composer yarn assets-dev perm] Install the project PHP and J
...
 
 clear-cache:    ## Clear the application cache in development
 	$(CONSOLE) cache:clear
 
+cc: clear-cache perm
```
Comment (tgalopin) [inline Makefile]: Same

File: Dockerfile (hunk 2 above)
Comment (rpg600) [inline Dockerfile]: Agreed, but it's easier to maintain when both containers share the same image.

Comment (mykiwi) [inline Dockerfile]: Both container don't share the same image.
```
app   => dev.dockerfile
tools => tools.dockerfile
```

Comment (rpg600) [inline Dockerfile]: Now there is 1 Dockerfile

**Comment by tgalopin** [comment]
Aren't we doing twice the job: https://github.com/EnMarche/en-marche.fr/pull/708 ?

**Comment by mykiwi** [comment]
#708 is a PR that can be merge into this PR

**Comment by mykiwi** [comment]
You may adapt the doc for using the new makefile.

File: docs/1-Installer-le-projet-en-local.md [hunk 6]
```diff
@@ -82,36 +82,36 @@ Vous devriez alors pouvoir lancer `make` qui vous affichera l'aide du Makefile :
...
+clean:           Clear and remove dependencies
+:---------------------------------------------------------------------------:
+test:            Run the PHP and the Javascript tests
+tu:              Run the PHP unit tests
+tf:              Run the PHP functional tests
+tj:              Run the Javascript tests
+lint:            Run lint on twig, yaml and Javascript files
+ls:              Lint Symfony (twig and yaml) files
+lj:              Lint the Javascript to follow the convention
+deps:            Install the project PHP and JS dependencies
```
Comment (HeahDude) [inline docs/1-Installer-le-projet-en-local.md]: Weird to have this at last of the test section.

Comment (rpg600) [inline docs/1-Installer-le-projet-en-local.md]: good catch

File: docs/1-Installer-le-projet-en-local.md [hunk 7]
```diff
@@ -82,36 +82,36 @@ Vous devriez alors pouvoir lancer `make` qui vous affichera l'aide du Makefile :
...
+start:           Install and start the project
+stop:            Remove docker containers
+reset:           Reset the whole project
+db:              Reset the database and load fixtures
+:---------------------------------------------------------------------------:
+db-diff:         Generate a migration by comparing your current database to your mapping information.
+watch:           Watch the assets and build their development version on change
+clear:           Remove all the cache, the logs, the sessions and the built assets
+clean:           Clear and remove dependencies
+deps:            Install the project PHP and JS dependencies
```
Comment (HeahDude) [inline docs/1-Installer-le-projet-en-local.md]: I would put it before `db` since it is a requirement for this section.

**Pull request merged**